src/util/baseutil.js

```diff
@@ -348,3 +348,46 @@ export function isEmpty(value) {
   }
   return true;
 }
+
+/**
+ * Check if IntersectionObserver is supported
+ * @return {boolean} true if window.IntersectionObserver is defined
+ */
+export function isIntersectionObserverSupported() {
+  // Check that 'IntersectionObserver' property is defined on window
+  return typeof window === "object" && window.IntersectionObserver;
+}
+
+/**
+ * Check if native lazy loading is supported
+ * @return {boolean} true if 'loading' property is defined for HTMLImageElement
+ */
+export function isNativeLazyLoadSupported() {
+  return typeof HTMLImageElement === "object" && HTMLImageElement.prototype.loading;
+}
+
+/**
+ * Calls onIntersect() when intersection is detected, or when
```

Comment:
Is the implementation for onIntersect() missing?

Reply:
onIntersect() is a callback passed to this function